clients/drcachesim/scheduler/scheduler.cpp

```diff
@@ -2966,7 +2966,7 @@ scheduler_tmpl_t<RecordType, ReaderType>::pick_next_input(output_ordinal_t outpu
                     if (prev_index == INVALID_INPUT_ORDINAL)
                         return eof_or_idle(output, need_lock, prev_index);
                     auto lock = std::unique_lock<std::mutex>(*inputs_[prev_index].lock);
-                    if (inputs_[prev_index].at_eof) {
+                    if (inputs_[prev_index].at_eof || inputs_[prev_index].unscheduled) {
```

Comment:
What is this change?  Is it a bug fix?  If so, probably should mention it in the PR description and/or spell out the situation it's covering in a line comment.

Reply:
Oh right, this was discovered by code inspection while figuring out the reason for some stats to be the way they are.  Not sure it was causing any visible issue: it is a particular case of an empty runqueue where if the current thread is supposed to be going unscheduled it looked like it might run again instead.  Added to PR description.

Reply:
Added comment here but it's pretty simple: "// If we can't go back to the current input, we're EOF or idle." when the runqueue is empty.